### PR TITLE
Mention NESTML/NEST Desktop in the section "Related projects" on the documentation main page

### PR DESCRIPTION
**Pull request by jd41**
Mentioning the aforementioned related projects in the Readthedocs index page, section "Related projects".

If there are further NEST-related projects worth mentioning, please add them!

Also added a subheading for License

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -103,12 +103,19 @@ Interested in contributing?
 * For more info about our larger community and the history of NEST check out the `NEST Initiative <https://www.nest-initiative.org>`_ website
 
 Related projects
-----------------
+################
+
+- `NESTML <https://github.com/nest/nestml>`_ - a domain specific language to describe neuron models in NEST
+- `NEST Desktop <https://github.com/babsey/nest-desktop>`_ - an interactive desktop application for NEST simulator
 
 The  `NeuralEnsemble <http://neuralensemble.org/>`_ is a community-based initiative to promote and co-ordinate open-source software development in neuroscience.
```
Comment (sarakonradi) [inline doc/index.rst:111]:
```suggestion
You may be interested in the associated tooling of NEST Simulator:

- `NESTML <https://nestml.readthedocs.io/en/latest/>`_ - a domain specific language to describe neuron models in NEST
- `NEST Desktop <https://nest-desktop.readthedocs.io/en/latest/>`_ - a web-based GUI application for NEST Simulator

```

File: doc/index.rst [hunk 2]
```diff
@@ -103,12 +103,19 @@ Interested in contributing?
...
-They host numerous software including `PyNN <http://neuralensemble.org/PyNN/>`_, a simulator-independent language for building neuronal network models and `Elephant (Electrophysiology Analysis Toolkit) <http://neuralensemble.org/elephant/>`_, a package for the analysis of neurophysiology data, using Neo data structures.
+They host numerous software including
 
+- `PyNN <http://neuralensemble.org/PyNN/>`_ - a simulator-independent language for building neuronal network models, and
+- `Elephant (Electrophysiology Analysis Toolkit) <http://neuralensemble.org/elephant/>`_ - a package for the analysis of neurophysiology data, using Neo data structures.
```
Comment (sarakonradi) [inline doc/index.rst:115]:
```suggestion
As well as:

- `PyNN <http://neuralensemble.org/PyNN/>`_ - a simulator-independent language for building neuronal network models
- `Elephant <http://neuralensemble.org/elephant/>`_ - a package for the analysis of neurophysiological data, using Neo data structures
```

File: doc/index.rst [hunk 3]
```diff
@@ -103,17 +103,22 @@ Interested in contributing?
 * For more info about our larger community and the history of NEST check out the `NEST Initiative <https://www.nest-initiative.org>`_ website
 
 Related projects
-----------------
+################
 
-The  `NeuralEnsemble <http://neuralensemble.org/>`_ is a community-based initiative to promote and co-ordinate open-source software development in neuroscience.
-They host numerous software including `PyNN <http://neuralensemble.org/PyNN/>`_, a simulator-independent language for building neuronal network models and `Elephant (Electrophysiology Analysis Toolkit) <http://neuralensemble.org/elephant/>`_, a package for the analysis of neurophysiology data, using Neo data structures.
+You may be interested in the associated tooling of NEST Simulator:
```
Comment (terhorstd) [inline doc/index.rst:108]:
```suggestion
Many other extensions and open-source tools related to the NEST Simulator are available. In particular the following packages may be of interest: 
```

File: doc/index.rst [hunk 4]
```diff
@@ -103,17 +103,22 @@ Interested in contributing?
...
 
+- `NESTML <https://nestml.readthedocs.io/en/latest/>`_ - a domain specific language to describe neuron models in NEST
+- `NEST Desktop <https://nest-desktop.readthedocs.io/en/latest/>`_ - a web-based GUI application for NEST Simulator
+- `PyNN <http://neuralensemble.org/PyNN/>`_ - a simulator-independent language for building neuronal network models
+- `Elephant <http://neuralensemble.org/elephant/>`_ - a package for the analysis of neurophysiological data, using Neo data structures
 
+This list does not intend to be complete. You can find more projects by the community by searching GitHub for the topics `"nest-simulator" <https://github.com/topics/nest-simulator>`_ or `"nest-module" <https://github.com/topics/nest-module/>`_.
```
Comment (terhorstd) [inline doc/index.rst:115]:
```suggestion
You can find more projects by the community by searching GitHub for the topics `"nest-simulator" <https://github.com/topics/nest-simulator>`_ or `"nest-module" <https://github.com/topics/nest-module/>`_.
```

File: doc/index.rst [hunk 5]
```diff
@@ -103,17 +103,22 @@ Interested in contributing?
...
+################
 
-The  `NeuralEnsemble <http://neuralensemble.org/>`_ is a community-based initiative to promote and co-ordinate open-source software development in neuroscience.
-They host numerous software including `PyNN <http://neuralensemble.org/PyNN/>`_, a simulator-independent language for building neuronal network models and `Elephant (Electrophysiology Analysis Toolkit) <http://neuralensemble.org/elephant/>`_, a package for the analysis of neurophysiology data, using Neo data structures.
+Many other extensions and open-source tools related to the NEST Simulator are available. In particular the following packages may be of interest: 
 
+- `NESTML <https://nestml.readthedocs.io/en/latest/>`_ - a domain specific language to describe neuron models in NEST
+- `NEST Desktop <https://nest-desktop.readthedocs.io/en/latest/>`_ - a web-based GUI application for NEST Simulator
+- `PyNN <http://neuralensemble.org/PyNN/>`_ - a simulator-independent language for building neuronal network models
+- `Elephant <http://neuralensemble.org/elephant/>`_ - a package for the analysis of neurophysiological data, using Neo data structures
```
Comment (sarakonradi) [inline doc/index.rst:113]:
```suggestion
- `NEST Desktop <https://nest-desktop.readthedocs.io/en/latest/>`_ - a web-based GUI application for NEST Simulator
- `NESTML <https://nestml.readthedocs.io/en/latest/>`_ - a domain specific language to describe neuron models in NEST
- `PyNN <http://neuralensemble.org/PyNN/>`_ - a simulator-independent language for building neuronal network models
- `Elephant <http://neuralensemble.org/elephant/>`_ - a package for the analysis of neurophysiological data, using Neo data structures
```

**Comment by terhorstd** [comment]
I opened an issue for the failing build on travis #1701. It is not related to this PR.

**Comment by sarakonradi** [comment]
#1623 fixes some of the Travis build failures. 

@jd41 Could you please ``git pull upstream master``?

**Pull request merged**